lxd/api.go

```diff
@@ -157,9 +166,14 @@ func restServer(d *Daemon) *http.Server {
 		w.Header().Set("Content-Type", "application/json")
 
 		ua := r.Header.Get("User-Agent")
-		if uiEnabled && strings.Contains(ua, "Gecko") {
-			// Web browser handling.
-			http.Redirect(w, r, "/ui/", http.StatusMovedPermanently)
+		if strings.Contains(ua, "Gecko") {
+			if !uiEnabled {
+				w.Header().Set("Content-Type", "text/html")
+				w.WriteHeader(http.StatusOK)
+				fmt.Fprint(w, errorMessage)
+			} else {
+				http.Redirect(w, r, "/ui/", http.StatusMovedPermanently)
+			}
```

Comment:
I refactor the code to do a redirection and it is fine for path `/ui/`
```suggestion
	      if strings.Contains(ua, "Gecko") {
			http.Redirect(w, r, "/ui/", http.StatusMovedPermanently)
			return
		} else {
			// Normal client handling.
			_ = response.SyncResponse(true, []string{"/1.0"}).Render(w)
		}
```
However, if a client enter `/ui`, or `/ui/login` or even `/foo` the same error is returned if the ui is not enabled
`{"type":"sync","status":"Success","status_code":200,"operation":"","error_code":0,"error":"","metadata":["/1.0"]}
`

Reply:
Actually the error is the same even when the UI is enabled and the client requests `/foo`
```
{"type":"sync","status":"Success","status_code":200,"operation":"","error_code":0,"error":"","metadata":["/1.0"]}
` ``